docs/advanced/framework_internals.md

```diff
@@ -0,0 +1,3 @@
+# Framework internals
```

Comment:
We should make a note on the usage of lark as well as how to contribute with new expressions workflow: 1) make the expression being recognized by lark and the work on the implementation - what we want to do when it recognizes it. Maybe this is planned for later, just wanted to make sure we reference that as well somewhere in the documentation.

Reply:
Yes, this is definitely the main topic for this page